### PR TITLE
Implement Bot API 6.0

### PR DESCRIPTION
**Pull request by tuscen**

### CODE REVIEW

File: src/Telegram.Bot/Requests/Inline Mode/AnswerInlineQueryRequest.cs [hunk 1]
```diff
@@ -85,4 +85,4 @@ public AnswerInlineQueryRequest(string inlineQueryId, IEnumerable<InlineQueryRes
         InlineQueryId = inlineQueryId;
         Results = results;
     }
-}
\ No newline at end of file
+}
```
Comment (karb0f0s) [inline src/Telegram.Bot/Requests/Inline Mode/AnswerInlineQueryRequest.cs:88]: `insert_final_newline = true` in .editorconfig?

File: src/Telegram.Bot/Requests/Abstractions/Documentation.cs [hunk 2]
```diff
@@ -78,4 +78,4 @@ internal abstract class Documentation
     /// Protects the contents of sent messages from forwarding and saving
     /// </summary>
     object ProtectContent;
-}
\ No newline at end of file
+}
```
Comment (karb0f0s) [inline src/Telegram.Bot/Requests/Abstractions/Documentation.cs:81]: `insert_final_newline = true` in .editorconfig?

File: src/Telegram.Bot/Requests/Available methods/Commands/DeleteMyCommandsRequest.cs [hunk 3]
```diff
@@ -6,9 +6,9 @@
 namespace Telegram.Bot.Requests;
 
 /// <summary>
-/// Use this method to delete the list of the bot's commands for the given
+/// Use this method to delete the list of the bot`s commands for the given
```
Comment (karb0f0s) [inline src/Telegram.Bot/Requests/Available methods/Commands/DeleteMyCommandsRequest.cs:9]: ` (backtick) is grammatically incorrect. it should be ' (tick) or ’ (apostrophe)

File: src/Telegram.Bot/Types/CallbackGame.cs [hunk 4]
```diff
@@ -4,9 +4,10 @@
 namespace Telegram.Bot.Types;
 
 /// <summary>
-/// A placeholder, currently holds no information. Use <see href="https://t.me/botfather">@BotFather</see> to set up your game.
+/// A placeholder, currently holds no information. Use <a href="https://t.me/botfather">@BotFather</a>
+/// to set up your game.
```
Comment (karb0f0s) [inline src/Telegram.Bot/Types/CallbackGame.cs:8]: add `max_line_length=xxx` in .editorconfig for consistency?

File: test/Telegram.Bot.Tests.Integ/Admin Bot/ChannelAdminBotTests.cs [hunk 5]
```diff
@@ -76,7 +76,7 @@ public async Task Should_Pin_Message()
         _classFixture.PinnedMessage = msg;
     }
 
-    [OrderedFact("Should get chat's pinned message")]
+    [OrderedFact("Should get chat`s pinned message")]
```
Comment (karb0f0s) [inline test/Telegram.Bot.Tests.Integ/Admin Bot/ChannelAdminBotTests.cs:79]: ` (backtick) is grammatically incorrect. it should be ' (tick) or ’ (apostrophe)

File: src/Telegram.Bot/Types/WebhookInfo.cs [hunk 6]
```diff
@@ -50,6 +50,13 @@ public class WebhookInfo
     [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
     public string? LastErrorMessage { get; set; }
 
+    /// <summary>
+    /// Optional. Unix time of the most recent error that happened when trying to synchronize available updates
+    /// with Telegram datacenters
+    /// </summary>
+    [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+    public DateTime? LastSynchronizationErrorDate { get; set; }
```
Comment (karb0f0s) [inline src/Telegram.Bot/Types/WebhookInfo.cs:58]: [JsonConverter(typeof(UnixDateTimeConverter))]

File: src/Telegram.Bot/Types/VoiceChatStarted.cs [hunk 7]
```diff
@@ -6,7 +7,7 @@ namespace Telegram.Bot.Types;
 /// <summary>
 /// This object represents a service message about a voice chat started in the chat. Currently holds no information.
 /// </summary>
+[Obsolete("This type will be removed in the next major version, use VoiceChatStarted instead")]
 [JsonObject(MemberSerialization.OptIn, NamingStrategyType = typeof(SnakeCaseNamingStrategy))]
 public class VoiceChatStarted
-{
-}
\ No newline at end of file
+{ }
```
Comment (karb0f0s) [inline src/Telegram.Bot/Types/VoiceChatStarted.cs:13]: add `csharp_empty_block_style = together ` rule to the `.editorconfig`?

File: src/Telegram.Bot/Types/VideoChatStarted.cs [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+using Newtonsoft.Json.Serialization;
+
+namespace Telegram.Bot.Types;
+
+/// <summary>
+/// This object represents a service message about a video chat started in the chat. Currently holds no information.
+/// </summary>
+[JsonObject(MemberSerialization.OptIn, NamingStrategyType = typeof(SnakeCaseNamingStrategy))]
+public class VideoChatStarted
+{ }
```
Comment (karb0f0s) [inline src/Telegram.Bot/Types/VideoChatStarted.cs:11]: add `csharp_empty_block_style  = together` rule to the `.editorconfig`?

File: src/Telegram.Bot/Types/VideoChatEnded.cs [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+namespace Telegram.Bot.Types;
+
+/// <summary>
+/// This object represents a service message about a video chat ended in the chat.
+/// </summary>
+[JsonObject(MemberSerialization.OptIn, NamingStrategyType = typeof(SnakeCaseNamingStrategy))]
+public class VideoChatEnded
+{
+    /// <summary>
+    /// Voice chat duration; in seconds
```
Comment (karb0f0s) [inline src/Telegram.Bot/Types/VideoChatEnded.cs:13]:
```c#
/// Voice chat duration in seconds
```

File: src/Telegram.Bot/Types/VideoChatEnded.cs [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    /// </summary>
+    [JsonProperty(Required = Required.Always)]
```
Comment (karb0f0s) [inline src/Telegram.Bot/Types/VideoChatEnded.cs:15]: why `Duration` field is `Required = Required.Always` vs `Users[]` is `DefaultValueHandling = DefaultValueHandling.Ignore` in `VideoChatParticipantsInvited`?

File: src/Telegram.Bot/Types/SentWebAppMessage.cs [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+/// <a href="https://core.telegram.org/bots/webapps">Web App</a> on behalf of a user.
+/// </summary>
+[JsonObject(MemberSerialization.OptIn, NamingStrategyType = typeof(SnakeCaseNamingStrategy))]
+public class SentWebAppMessage
+{
+    /// <summary>
+    /// Optional. Identifier of the sent inline message. Available only if there is an inline keyboard attached
+    /// to the message.
+    /// </summary>
+    [JsonProperty(Required = Required.Always)]
```
Comment (karb0f0s) [inline src/Telegram.Bot/Types/SentWebAppMessage.cs:17]:
```c#
[JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
```

File: src/Telegram.Bot/Types/SentWebAppMessage.cs [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+    public string InlineMessageId { get; set; } = default!;
```
Comment (karb0f0s) [inline src/Telegram.Bot/Types/SentWebAppMessage.cs:18]:
```c#
public string? InlineMessageId { get; set; };
```

File: src/Telegram.Bot/Types/Message.cs [hunk 13]
```diff
@@ -456,6 +490,11 @@ this switch
             { VoiceChatStarted: { } }              => MessageType.VoiceChatStarted,
             { VoiceChatEnded: { } }                => MessageType.VoiceChatEnded,
             { VoiceChatParticipantsInvited: { } }  => MessageType.VoiceChatParticipantsInvited,
+            { VideoChatScheduled: { } }            => MessageType.VideoChatScheduled,
```
Comment (karb0f0s) [inline src/Telegram.Bot/Types/Message.cs:493]: wouldn't switch process deprecated `Voice*` properties before and thus `Video*` message type will never be populated?

File: src/Telegram.Bot/Types/VideoChatEnded.cs (hunk 10 above)
Comment (tuscen) [reply src/Telegram.Bot/Types/VideoChatEnded.cs:15]: It's a but from previous versions, will fix it

File: src/Telegram.Bot/Types/Message.cs (hunk 13 above)
Comment (tuscen) [reply src/Telegram.Bot/Types/Message.cs:493]: No, MessageType is just for convenience and isn't present in Bot API. The actual property will be deserialized, it's just for some time both Voice* and Video* properties will be populated, but Message.Type will tell users that it has Voice* property populated. After Telegram stops send Voice* properties only Video* props will be populated and Message.Type will tell users that it's Video* type of a message

File: src/Telegram.Bot/Requests/Abstractions/Documentation.cs (hunk 2 above)
Comment (tuscen) [reply src/Telegram.Bot/Requests/Abstractions/Documentation.cs:81]: Already present, just forgot to enable it in Rider to enforce it.

**Review by karb0f0s — APPROVED** [review]

File: src/Telegram.Bot/Types/Message.cs (hunk 13 above)
Comment (karb0f0s) [reply src/Telegram.Bot/Types/Message.cs:493]: I do not agree. Either we should put `VideoChat*` pattern before `VoiceChat*` or we should not yield `VoiceChat*` message type at all - it's misleading.

File: test/Telegram.Bot.Tests.Unit/Serialization/MenuButtonSerializationTests.cs [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+        {
+            WebApp = new WebAppInfo()
+            {
+                Url = "https://example.com/link/to/web/app"
+            },
+            Text = "Test text"
+        };
+
+        string? webAppButtonJson = JsonConvert.SerializeObject(webAppButton);
+        Assert.Contains(@"""type"":""web_app""", webAppButtonJson);
```
Comment (karb0f0s) [inline test/Telegram.Bot.Tests.Unit/Serialization/MenuButtonSerializationTests.cs:49]: I wish we could use [raw-string-literals](
https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-11#raw-string-literals)

**Review by karb0f0s — APPROVED** [review]
LGTM

**Pull request merged**